### ISSUE TITLE
type check for then, andThen, etc. should allow undefined?

### ISSUE THREAD

**Issue by nfcampos** [heroku/react-refetch#126]
Not allowing undefined means I can't do this:
```js
connect(() => ({
  patch: (body, bar) => ({
    foo: {
      url: '/foo'
      andThen: bar ? ({
        bar: '/bar'
      }) : undefined
  })
}))
```

I have to assign the mapping to a variable and then conditionally assign `mapping.andThen`. This doesn't really make sense because accessing a property that wans't defined returns `undefined` anyway.
what do you think @ryanbrainard ?

**Comment by ryanbrainard** [comment]
I can't remember offhand what it was, but I was also running into some kind of problem with these not being able to be undefined. I think we should allow it.

**Comment by nfcampos** [comment]
alright, should we allow undefined for all keys or just these 4?

**Comment by ryanbrainard** [comment]
Hmm, good question. `undefined` does not make sense for some of them where a value is required (e.g. `method`). I guess we could make `undefined` assign the default in those cases, but I just don't want it to be too magical. I think we should do whatever is the most consistent for the user. I'm a bit on the fence, but I think we should just do it for these 4 for now and consider doing the others later if it makes sense. Do you have any opinions on it?

**Comment by nfcampos** [comment]
yeah I think we can allow undefined only for these 4 for now

**Comment by nfcampos** [comment]
closed by #128
